packages/@okta/vuepress-site/docs/guides/oie-embedded-sdk-use-cases/oie-embedded-sdk-use-case-pwd-recovery-mfa/go/integrationsteps.md

```diff
@@ -0,0 +1,167 @@
+
+```go
+rpr, err = rpr.VerifyEmail(context.TODO())
+if err != nil {
+ //Handle error
+}
+
+if !rpr.HasStep(idx.ResetPasswordStepEmailConfirmation) {
+ //This is an error case.
+}
+ //Store the response in cache
+s.cache.Set("resetPasswordFlow", rpr, time.Minute*5)
+```
+
+### Step 4: Show the code verification page
+
+The next step is for the user to enter the verification code from their email.
+Build a page that captures this code.
+
+<div class="common-image-format common-image-format-vertical-margin">
+
+![Screenshot of second page in the password recovery flow showing the verification code field](/img/oie-embedded-sdk/oie-embedded-sdk-go-use-case-pwd-recover-page-second.png)
+
+</div>
+
+### Step 5: Submit the verification code
+
+When the user submits the verification code, call the `ResetPasswordResponse` object's
+`ConfirmEmail` method, passing in the verification code.
+
+```go
+ //Get session store so we can store our tokens
+session, err := sessionStore.Get(r, "direct-auth")
+
+rpr, err = rpr.ConfirmEmail(context.TODO(), r.FormValue("code"))
+if err != nil {
+ //Error handling
+}
+
+if !rpr.HasStep(idx.ResetPasswordStepNewPassword) {
+ //Error condition
+}
+
+s.cache.Set("resetPasswordFlow", rpr, time.Minute*5)
+
+http.Redirect(w, r, "/passwordRecovery/newPassword", http.StatusFound)
+return
+```
+
+### Step 6: Show the new password page
+
+After the code has been confirmed, the next step is to send the user
+to a page that allows them to enter their new password.
+
+<div class="common-image-format common-image-format-vertical-margin">
+
+![Screenshot of third page in the password recovery flow showing the password fields](/img/oie-embedded-sdk/oie-embedded-sdk-go-use-case-pwd-recover-page-third.png)
+
+</div>
+
+### Step 7: Submit the new password
+
+After the user enters their password and submits, call the `ResetPasswordResponse` object's
+`SetNewPassword` method to change their password to the new password. If successful,
+this method should return the sign-in tokens.
+
+```go
+rpr := tmp.(*idx.ResetPasswordResponse)
+
+rpr, err = rpr.SetNewPassword(context.TODO(), newPassword)
+if err != nil {
+ //Error handling
+}
+
+if !rpr.HasStep(idx.ResetPasswordStepSuccess) {
+ //Error handling
+}
+```
+
+### Step 8: Store the tokens in a session and redirect to the signed-in home page
+
+Store the tokens from the `ResetPasswordResponse` into session
+for later use. After the tokens are stored, redirect the user to the
+default signed-in home page.
+
+```go
+ //If we have tokens we have success, so let's store tokens
+if rpr.Token() != nil {
+  session.Values["access_token"] = rpr.Token().AccessToken
+  session.Values["id_token"] = rpr.Token().IDToken
+  err = session.Save(r, w)
+  if err != nil {
+    log.Fatalf("could not save access token: %s", err)
+  }
+} else {
+ //Error handling
+}
+
+ //Redirect the user to /profile
+http.Redirect(w, r, "/", http.StatusFound)
+return
+```
+
+### Step 9 (Optional) Retrieve user profile information
```

Comment:
This needs a colon after "9", for consistency with all the other headings.

Reply:
Colon added.